python/receipt_printer.py

```diff
@@ -0,0 +1,56 @@
+from model_objects import ProductUnit
+
+class ReceiptPrinter:
+
+    def __init__(self, columns=40):
+        self.columns = columns
+  
+    def print_receipt(self, receipt):
+        result = ""
+        for item in receipt.items:
+            receipt_item = self.print_receipt_item(item)
+            result += receipt_item
+
+        for discount in receipt.discounts:
+            discount_presentation = self.print_discount(discount)
+            result += discount_presentation
+
+        result += "\n"
+        result += self.present_total(receipt)
+        return str(result)
+
+    def print_receipt_item(self, item):
+        total_price_printed = self.print_price(item.total_price)
+        name = item.product.name
+        line = self.format_line_with_whitespace(name, total_price_printed)
+        if item.quantity != 1:
+            line += f"  {self.print_price(item.price)} * {self.print_quantity(item)}\n"
+        return line
+
+    def format_line_with_whitespace(self, name, value):
```

Comment:
```suggestion
    def build_line_with_whitespace(self, name, value):
```
This does not format line, but build a new line.

Reply:
what are parameters `name` and `value` (for a method which speak only about line with whitespace)?

Reply:
It seems that this method should be used each time we add a new line in the printing.
What about this?

- a new class `SalesReceipt` with
- private attribute with the texte currently printing
- `def add_line(title, value)`